service/src/test/java/de/adorsys/psd2/sandbox/xs2a/pis/PaymentInitiationWithScaSteps.java

```diff
@@ -86,10 +126,78 @@ public void initiatePayment(String paymentProduct) {
     context.setPaymentId(response.getBody().getPaymentId());
   }
 
+  @Given("^PSU initiated a periodic payment using the payment product (.*)$")
+  public void initiatePeriodicPayment(String paymentProduct) {
+    context.setPaymentService("periodic-payments");
+    context.setPaymentProduct(paymentProduct);
+
+    HashMap<String, String> headers = TestUtils.createSession();
+
+    PeriodicPaymentInitiationSctJson periodicPayment = new PeriodicPaymentInitiationSctJson();
+    periodicPayment.setEndToEndIdentification("WBG-123456789");
+
+    HashMap<String, String> debtorAccount = createAccount("DE51250400903312345678");
+    periodicPayment.setDebtorAccount(debtorAccount);
+
+    Amount instructedAmount = new Amount();
+    instructedAmount.setAmount("520");
+    instructedAmount.setCurrency("EUR");
+    periodicPayment.setInstructedAmount(instructedAmount);
+
+    HashMap<String, String> creditorAccount = createAccount("DE15500105172295759744");
+    periodicPayment.setCreditorAccount(creditorAccount);
+
+    periodicPayment.setCreditorAgent("AAAADEBBXXX");
+    periodicPayment.setCreditorName("WBG");
+    periodicPayment.setFrequency(FrequencyCode.MONTHLY);
+    periodicPayment.setDayOfExecution(DayOfExecution._01);
+    periodicPayment.setExecutionRule(ExecutionRule.PRECEEDING);
+    periodicPayment.setStartDate(LocalDate.of(2019,01,20));
+    periodicPayment.setEndDate(LocalDate.of(2020,01,20));
+
+    Address creditorAddress = createCreditorAddress();
+    periodicPayment.setCreditorAddress(creditorAddress);
+
+    periodicPayment.setRemittanceInformationUnstructured("Ref. Number WBG-1222");
+
+    Request<PeriodicPaymentInitiationSctJson> request = new Request<>();
+    request.setBody(periodicPayment);
+    request.setHeader(headers);
+
+    ResponseEntity<PaymentInitationRequestResponse201> response = template.exchange(
+        context.getPaymentService() + "/" +
+            context.getPaymentProduct(),
+        HttpMethod.POST,
+        request.toHttpEntity(),
+        PaymentInitationRequestResponse201.class);
+
+    context.setPaymentId(response.getBody().getPaymentId());
+  }
+
+
+
+  @NotNull
+  private HashMap<String, String> createAccount(String iban) {
+    HashMap<String, String> creditorAccount = new HashMap<>();
+    creditorAccount.put("currency", "EUR");
```

Comment:
In this case it would make sense to also pass the `Currency` otherwise rename the method to something like `createEuroAccount`

Reply:
done